grug/app/src/execute.rs

```diff
@@ -777,16 +777,22 @@ where
 
 // ----------------------------------- cron ------------------------------------
 
-pub fn do_cron_execute<VM>(ctx: AppCtx<VM>, contract: Addr) -> AppResult<Vec<Event>>
+pub fn do_cron_execute<VM>(mut ctx: AppCtx<VM>, contract: Addr) -> AppResult<Vec<Event>>
 where
     VM: Vm + Clone,
     AppError: From<VM::Error>,
 {
+    let buffer = Shared::new(Buffer::new(ctx.storage, None));
+
+    ctx.storage = Box::new(buffer.clone());
```

Comment:
stuff like this is why i decided to get rid of `AppCtx`. mutating `ctx.storage` just feel wrong... i.e. an anti-pattern